go/vt/topo/etcd2topo/lock.go

```diff
@@ -109,101 +108,81 @@ func (s *Server) waitOnLastRev(ctx context.Context, nodePath string, revision in
 	return false, nil
 }
 
-func (s *Server) lock(ctx context.Context, nodePath string, contents string) (string, error) {
+// etcdLockDescriptor implements topo.LockDescriptor.
+type etcdLockDescriptor struct {
+	c       *cellClient
+	leaseID clientv3.LeaseID
+}
+
+// Lock is part of the topo.Backend interface.
+func (s *Server) Lock(ctx context.Context, cell, dirPath, contents string) (topo.LockDescriptor, error) {
+	// We list the directory first to make sure it exists.
+	if _, err := s.ListDir(ctx, cell, dirPath); err != nil {
+		if err == topo.ErrNoNode {
+			return nil, err
+		}
+		return nil, fmt.Errorf("cannot ListDir(%v,%v) before locking", cell, dirPath)
+	}
+
+	return s.lock(ctx, cell, dirPath, contents)
+}
+
+// lock is used by both Lock() and master election.
```

Comment:
Mind removing this non-useful comment as well?

Reply:
This one is useful, it justifies why this is just not directly inside the Lock method.